### ISSUE TITLE
Spurious math errors in shaders for ESSL shaders

### ISSUE THREAD

**Issue by kwokcb** [AcademySoftwareFoundation/MaterialX#992]
There appears to be errors occurring for some shaders, which can be seen when running the Web MaterialXView.
e.g.
```
C:\fakepath(645,50-70): warning X4008: floating point division by zero
C:\fakepath(645,24-108): warning X4118: imaginary square root
C:\fakepath(573,32-40): warning X4008: floating point division by zero
C:\fakepath(582,25-60): warning X4118: imaginary square root
C:\fakepath(645,50-70): warning X4008: floating point division by zero
C:\fakepath(645,24-108): warning X4118: imaginary square root
C:\fakepath(573,32-40): warning X4008: floating point division by zero
C:\fakepath(582,25-60): warning X4118: imaginary square root
C:\fakepath(645,50-70): warning X4008: floating point division by zero
C:\fakepath(645,24-108): warning X4118: imaginary square root
C:\fakepath(573,32-40): warning X4008: floating point division by zero
C:\fakepath(582,25-60): warning X4118: imaginary square root
C:\fakepath(645,50-70): warning X4008: floating point division by zero
C:\fakepath(645,24-108): warning X4118: imaginary square root
C:\fakepath(573,32-40): warning X4008: floating point division by zero
C:\fakepath(582,25-60): warning X4118: imaginary square root
```
This does not appear to be geometry related as this occurs with different geometries.

These should probably be looked at. They do not appear to occur for the default std surface shader.
e.g. Load in the glass shader example shader.

**Comment by jstone-lucasfilm** [comment]
Thanks for this report, @kwokcb, and I believe these issues are addressed by pull request #1019.  I'd like to get feedback from Niklas on this proposal, though, before moving forward.

**Comment by jstone-lucasfilm** [comment]
Merged pull request #1019
